packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +17,75 @@ declare global {
 /**
  * A nimble styled rich text viewer
  */
-export class RichTextViewer extends FoundationElement {}
+export class RichTextViewer extends FoundationElement {
+    /**
+     * @public
+     * @remarks
+     * Accessor: uses the parsing logic to convert the input markdown string to render in the component.
+     */
+    public set markdownValue(value: string) {
+        this._markdownValue = value;
+        this.serializedContent = this.parseMarkdownToDOM(value);
+    }
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: gets the raw markdown string.
+     */
+    public get markdownValue(): string {
+        return this._markdownValue;
+    }
+
+    public serializedContent?: HTMLElement | DocumentFragment;
+    private _markdownValue = '';
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.appendSerializedContentToViewer();
+    }
+
+    /**
+     *
+     * This function takes a markdown string, parses it using the ProseMirror MarkdownParser, serializes the parsed content into a
+     * DOM structure using a DOMSerializer, and returns the serialized result.
+     *
+     * @internal
+     */
+    public parseMarkdownToDOM(value: string): HTMLElement | DocumentFragment {
```

Comment:
The initial thought of having this as `public` is that we could use the same function for the `nimble-rich-text-editor` when parsing the markdown string to load it in the editor. But it would be better not to make inter-communication between two different components. 

So as you suggested I have made this function `private` and removed one of the test cases which checks if this function is called or not when setting the markdown input to the component.